config/opal_check_attributes.m4

```diff
@@ -0,0 +1,611 @@
+dnl -*- shell-script -*-
```

Comment:
It seems like we should be fine replacing the new `opal_check_attributes.m4` with the old script.  I've made that change.